### PR TITLE
More helpful contact page

### LINKED ISSUE #369: Improve Contact page
https://www.writetothem.com/about-contact

fireshot screen capture 1013 - writetothem - contact technical support - www_writetothem_com_about-contact

If a user selects 'your representative', shouldn't a big error message come up to tell them to go back to the homepage and put in their postcode?

Emails intended for MPs are still one of the most frequent user support tasks. In a Slack chat yesterday, it was suggested that we could equate the amount of time spent on replying to these users to the amount of time we might dedicate to fixing the problem, but I don't think that's quite right: 
a) If we don't fix it, the problem is ongoing, forever
b) users won't always return to WTT and start again when we write back and tell them to, and anyway we've introduced a gap of up to 72 hours (if it happens to span a weekend) between what they intended to do and our telling them they got it wrong
c) It leaves a negative feeling of having done something stupid with the user, and this will rub off on their feelings about the site

ref #21

### PR DESCRIPTION
**Pull request by zarino**
Fixes #369.

Branched off the `309-development-environment` branch from #464.

~~@dracos I notice, in addition to the regular contact page at `about-contact.html`, there’s an `about-contactresponse.html` file. Do you know where this gets used?~~

Screenshot_2021-02-10 WriteToThem

Screenshot_2021-02-10 WriteToThem(1)

Screenshot_2021-02-12 WriteToThem(1)

Screenshot_2021-02-12 WriteToThem

### CODE REVIEW

**Comment by dracos** [comment]
It's the page when you submit the form on the contact page.

**Comment by zarino** [comment]
@dracos I think this is ready for review now!

File: phplib/emailform.php [hunk 1]
```diff
@@ -94,83 +77,77 @@ function fyr_display_emailform () {
...
-  $value='';
+  $value = '';
   if (isset($defs['value']))
       $value = $defs['value'];
   if (isset($_POST[$defs['inputname']]))
       $value = $_POST[$defs['inputname']];
+
   $htmlvalue = htmlentities($value, ENT_QUOTES, 'UTF-8');
+  $class = isset($messages[$defs['inputname']]) ? 'error' : '';
+  $required = (isset($defs['required']) && $defs['required']) ? ' required' : '';
```
Comment (dracos) [inline phplib/emailform.php:88]: As we are on PHP7 now, you can use `($defs['required'] ?? 0)` here.

**Review by dracos — APPROVED** [review]
One tiny suggestion/new feature; only other issue I saw is the focus outline on the buttons if you tab goes through the first line of text underneath also, iyswim, dunno if that's easy to fix. But apart from that, looks good!

**Comment by zarino** [comment]
@dracos Fixed! Happy for all of this to be merged into `master` (including @sagepe’s earlier Docker commits)? Or should I rebase those out?

**Comment by sagepe** [comment]
There is a separate PR for the Docker commits (#464), so perhaps we should treat them separately 🤷🏻 

Also, if you (@zarino ) have used the environment, perhaps you'd be better placed to complete that review instead of @wrightmartin?

**Comment by zarino** [comment]
Thanks @sagepe. Have left a review on https://github.com/mysociety/writetothem/pull/464 :+1:

**Pull request merged**